docs/en/query_language/agg_functions/reference.md

```diff
@@ -906,32 +906,510 @@ SELECT quantileTiming(0.5)(number / 2) FROM numbers(10)
 
 A hash table is used as the algorithm. Because of this, if the passed values ​​are frequently repeated, the function consumes less RAM than `quantileExact`. You can use this function instead of `quantileExact` and specify the weight as 1.
 
-## quantileTDigest(level)(x)
+## quantileTDigest(level)(x) {#quantiletdigest}
 
 Approximates the quantile level using the [t-digest](https://github.com/tdunning/t-digest/blob/master/docs/t-digest-paper/histo.pdf) algorithm. The maximum error is 1%. Memory consumption by State is proportional to the logarithm of the number of passed values.
 
 The performance of the function is lower than for `quantile` or `quantileTiming`. In terms of the ratio of State size to precision, this function is much better than `quantile`.
 
 The result depends on the order of running the query, and is nondeterministic.
 
-## median(x)
+## median {#median}
 
-All the quantile functions have corresponding median functions: `median`, `medianDeterministic`, `medianTiming`, `medianTimingWeighted`, `medianExact`, `medianExactWeighted`, `medianTDigest`. They are synonyms and their behavior is identical.
+Computes median or 2-quantile of numeric data set. Median is a middle number of sequence, that separates higher half from the lower half of this sequence. If number of values in the data set is odd then mediana usually is an arithmetic mean of two middle values.
+
+This function based on [reservoir sampling](https://en.wikipedia.org/wiki/Reservoir_sampling) method with a reservoir size up to 8192. Since this algorithm uses a random number generator this function is non-deterministic.
+
+This algorithm provides very low accuracy
+
+**Syntax** 
+
+```sql
+median(x);
+```
+
+Alias: `quantile`. 
+
+Median is an alias of `quantile` function with an argument of 0.5 (50%).
+
+**Parameters** 
+
+- `x` — numeric data types, Date and DateTime.
+
+**Returned value**
+
+The middle value will be returned from the set with an odd number of values. For the set with an even number of values arithmetic mean of two middle values will be returned without any rounding.  
+
+Type: Float64, UInt64 and Int64 for numeric, Date and DateTime respectfully for corresponding data types.
+
+**Example**
+
+Input table:
+
+```text
+┌─val─┐
+│   1 │
+│   1 │
+│   2 │
+│   3 │
+└─────┘
+```
+
+Query:
+
+```sql
+SELECT median(val) FROM table
+```
+
+Result:
+
+```text
+┌─median(val)─┐
+│         1.5 │
+└─────────────┘
+```
+
+**See Also**
+
+- [quantile](#agg_function-quantile)
+
+## medianDeterministic {#medianDeterministic}
+
+Computes median or 2-quantile of numeric data set. This function uses [reservoir sampling](https://en.wikipedia.org/wiki/Reservoir_sampling) with a reservoir size up to 8192. 
+Function is deterministic due to usage of hash of passed number - the "determinator" - in the algorithm.
```

Comment:
This sentence is too vague for reader who don't know the signature of a function. Make a new header "Additional information" after Syntax and describe all the "deterministic" part there.